src/adaptors/tcp/tcp_adaptor.c

```diff
@@ -159,6 +165,21 @@ static void qdr_tcp_connection_ingress_accept(qdr_tcp_connection_t *tc);
 static void handle_outgoing(qdr_tcp_connection_t *conn);
 static void encrypt_outgoing_tls(qdr_tcp_connection_t *conn, qd_adaptor_buffer_t *unencrypted_buff, bool write_buffers);
 
+static void get_alpn_protocol(qdr_tcp_connection_t *conn, char **alpn_protocol)
```

Comment:
Can this logic be provided by a new API call to adaptor_tls rather than a tcp-only internal function? Seems like having a general "qd_tls_get_aplpn_protocol()" would be a useful.

Reply:
or better yet "qd_tls_get_alpn_protocol()"

Reply:
Done ! thanks